src/bin/cli/snapshot/mod.rs

```diff
@@ -0,0 +1,60 @@
+use catalyst_toolbox::snapshot::{RawSnapshot, Snapshot};
+use chain_addr::Discrimination;
+use jcli_lib::utils::{
+    output_file::{Error as OutputFileError, OutputFile},
+    output_format::{Error as OutputFormatError, OutputFormat},
+};
+use jormungandr_lib::interfaces::Stake;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use thiserror::Error;
+
+/// Process raw registrations into blockchain initials
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct SnapshotCmd {
+    /// Path to the file containing all CIP-15 compatible registrations in json format.
+    #[structopt(short, long, parse(from_os_str))]
+    snapshot: PathBuf,
+    /// Registrations voting power threshold for eligibility
+    #[structopt(short, long)]
+    threshold: Stake,
+
+    /// Discrimination to use for initial addresses
+    #[structopt(short, long)]
+    discrimination: Discrimination,
+
+    #[structopt(flatten)]
+    output: OutputFile,
+
+    #[structopt(flatten)]
+    output_format: OutputFormat,
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    OutputFile(#[from] OutputFileError),
+    #[error(transparent)]
+    OutputFormat(#[from] OutputFormatError),
+}
+
+impl SnapshotCmd {
+    pub fn exec(self) -> Result<(), Error> {
+        let raw_snapshot: RawSnapshot = serde_json::from_reader(File::open(&self.snapshot)?)?;
+        let initials = Snapshot::from_raw_snapshot(raw_snapshot, self.threshold)
+            .to_block0_initials(self.discrimination);
```

Comment:
What vitup expects is that root of json is initials, like below:

```
{
    "initial": [
        {
            "fund": [
                {
```

But this is not a big problem. However there is another discrepancy as funds collections are split by 100 items. Question if not splitting is ok? i remember that there was some upper limit of items in fund

Reply:
I guess that vitup expects a `Block0Configuration`, while here the only thing provided is the list of initials, so I would keep it that way to make it explicit that this tools only provides that.

For the second issue, I think that has to do with limits in fragment size, i.e. it's not logically wrong to have all fragments in a single `Initial::Fund`, but only 255 will fit in a fragment on chain. Tbh, we do implement `impl TryFrom<Initial> for Fragment`, so there is some expectation that you should be able to convert between the two.

If it's not too much work on vitup, I would suggest we keep this tool as blockchain agnostic as possible, and change this to return just a `Vec<InitialUTxO>` (i.e. just a collection of inital funds), leaving splitting to deployment tools.

Reply:
Agree!